plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -309,6 +325,32 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
 
             override val declarationSource: KtSourceElement? = symbol.source
         }
+
+        val declaration = symbol.fir as? FirSimpleFunction ?: return signature
+        val depth = 0
+        val returnTarget = ReturnTarget(depth, signature.callableType.returnType)
+        val stmtCtx = signature.createStatementContext(returnTarget, depth)
+        return stmtCtx.enhanceWithUserSpecifications(declaration, signature)
```

Comment:
Is this wrapper worth it?  It feels to me like it would be clearer to get the preconditions done here already.  Some comments on why things happen in a certain order would be good, I think.